velox/serializers/tests/PrestoSerializerTest.cpp

```diff
@@ -147,6 +147,22 @@ TEST_F(PrestoSerializerTest, basic) {
   testRoundTrip(rowVector);
 }
 
+TEST_F(PrestoSerializerTest, metadataOnlyVector) {
+  vector_size_t numRows = 1'000;
+  auto vector = vectorMaker_->rowVector(ROW({"a"}, {BIGINT()}), numRows);
+  auto rowVector = vectorMaker_->rowVector({vector});
+  std::ostringstream out;
+  serialize(rowVector, &out, nullptr);
+
+  auto rowType = ROW({}, {});
```

Comment:
This is not a valid usage pattern. The row type used to serialize must match row type used to deserialize.

Reply:
+1.

PrestoSerializer wire format is externally documented and should follow the semantics outlined in https://prestodb.io/docs/current/develop/serialized-page.html

An alternative here could be to read the number of columns from the next byte (https://github.com/facebookincubator/velox/blob/main/velox/serializers/PrestoSerializer.cpp#L1737) instead and validate if that matches what is expected by the type. We could throw an error on mismatch. That way we would understand there is a schema mismatch.

@mbasmanova, @xiaoxmeng : wdyt ?